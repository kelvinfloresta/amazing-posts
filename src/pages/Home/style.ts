import styled from "styled-components";

export const HomeTitle = styled.h1`
  color: ${(props) => props.theme.textColor1};
  text-align: center;
  margin: 3rem 0;
`;

export const FilterContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;
  height: 100px;
`;
