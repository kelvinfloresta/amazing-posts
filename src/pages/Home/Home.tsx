import React, { Fragment, useState } from "react";
import useListPost from "containers/useListPost";
import ListPost from "components/ListPost/ListPost";
import { HomeTitle, FilterContainer } from "./style";
import { useHistory } from "react-router-dom";
import Input from "components/Input/Input";

export default function Home() {
  const history = useHistory();
  const [userId, setUserId] = useState<number | undefined>();
  const { listPostError, listPostLoading, posts, retryListPost } = useListPost(
    userId
  );

  function seePost(postId: number) {
    history.push("/" + postId);
  }

  function onChangeUserId(e: React.ChangeEvent<HTMLInputElement>) {
    const userId = Number(e.target.value);
    setUserId(userId || undefined);
  }

  return (
    <Fragment>
      <HomeTitle>The amazing posts</HomeTitle>

      <FilterContainer>
        <label>Search for User using ID</label>
        <Input
          onChange={onChangeUserId}
          inputMode="numeric"
          type="number"
          placeholder="Ex: 1"
        />
      </FilterContainer>

      <ListPost
        isLoading={listPostLoading}
        posts={posts}
        retry={retryListPost}
        error={listPostError}
        onClickPost={seePost}
      />
    </Fragment>
  );
}
