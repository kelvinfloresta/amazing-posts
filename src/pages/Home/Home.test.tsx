import React from "react";
import renderer from "react-test-renderer";
import Home from "./Home";
import Post from "entities/Post.entity";

jest.mock("containers/useListPost", () => {
  const posts: Post[] = [{ id: 1, title: "title", body: "body" }];
  return () => {
    return {
      listPostLoading: false,
      posts,
    };
  };
});

describe("ViewPost", () => {
  it("should render correctly", () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
