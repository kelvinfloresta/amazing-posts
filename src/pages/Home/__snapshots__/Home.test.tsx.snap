// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewPost should render correctly 1`] = `
Array [
  <h1
    className="sc-fzozJi flsGJq"
  >
    The amazing posts
  </h1>,
  <div
    className="sc-fzoLsD haoCQg"
  >
    <label>
      Search for User using ID
    </label>
    <input
      className="sc-fzpans jFhoPn"
      inputMode="numeric"
      onChange={[Function]}
      placeholder="Ex: 1"
      type="number"
    />
  </div>,
  <div
    className="sc-AxjAm gMyFWH"
  >
    <div
      className="sc-AxirZ dVvgfn"
    >
      <div
        className="sc-AxiKw hcqcwH"
        height="200px"
        onClick={[Function]}
      >
        <h4
          className="sc-AxhCb fbvUGD"
          title="title"
        >
          title
        </h4>
        <p
          className="sc-AxhUy loiAMx"
        >
          body
        </p>
      </div>
    </div>
  </div>,
]
`;
